x-pack/plugins/security/server/routes/telemetry/authentication_type.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { createHash } from 'crypto';
+
+import { schema } from '@kbn/config-schema';
+
+import type { RouteDefinitionParams } from '..';
+import { wrapIntoCustomErrorResponse } from '../../errors';
+import { createLicensedRouteHandler } from '../licensed_route_handler';
+
+const COUNTER_TYPE = 'SecurityAuthType';
+
+export function defineTelemetryOnAuthTypeRoutes({
+  getAuthenticationService,
+  router,
+  usageCounter,
+}: RouteDefinitionParams) {
+  router.post(
+    {
+      path: '/internal/security/telemetry/auth_type',
+      validate: {
+        body: schema.nullable(
+          schema.object({
+            auth_type: schema.string(),
+            timestamp: schema.number(),
+            username_hash: schema.string(),
+          })
+        ),
+      },
+    },
+    createLicensedRouteHandler(async (context, request, response) => {
+      try {
+        let timestamp = new Date().getTime();
+        const {
+          auth_type: oldAuthType,
+          username_hash: oldUsernameHash,
+          timestamp: oldTimestamp,
+        } = request.body || {
+          auth_type: '',
+          username_hash: '',
+          timestamp,
+        };
+        const authUser = await getAuthenticationService().getCurrentUser(request);
+        const usernameHash = authUser
+          ? createHash('sha3-256').update(authUser?.username).digest('hex')
+          : '';
+
+        const elapsedTimeInHrs = (timestamp - oldTimestamp) / (1000 * 60 * 60);
+
+        if (
+          !authUser?.authentication_type ||
+          (authUser?.authentication_type && authUser?.authentication_type === '')
+        ) {
+          return response.badRequest({
+            body: { message: `Authentication type can not be empty` },
+          });
+        }
+
+        if (
+          usageCounter &&
+          (elapsedTimeInHrs >= 12 ||
+            oldUsernameHash !== usernameHash ||
+            oldAuthType !== authUser?.authentication_type)
+        ) {
+          usageCounter.incrementCounter({
+            counterName: authUser?.authentication_type,
```

Comment:
The source issue (#117517) suggests recording more data points:

> If an interactive session is detected, then we want to record the following from the [`AuthenticatedUser` model](https://github.com/elastic/kibana/blob/6ecffcc57c9a5b0328257cd25b737140a8471dda/x-pack/plugins/security/common/model/authenticated_user.ts#L21):
> * `authentication_realm.type`
> * `lookup_realm.type`
> * `authentication_provider.type`
> * `authentication_type`

I might not be in the loop though. Did you intentionally leave the the other ones out, or was that an oversight?

Reply:
I am not even sure, let's talk to @legrego about it when he is back. I will put back the PR in progress to avoid slack/notification.